pkg/api/plugins.go

```diff
@@ -492,18 +491,6 @@ func (hs *HTTPServer) UninstallPlugin(c *contextmodel.ReqContext) response.Respo
 }
 
 func translatePluginRequestErrorToAPIError(err error) response.Response {
-	if errors.Is(err, backendplugin.ErrPluginNotRegistered) {
-		return response.Error(http.StatusNotFound, "Plugin not found", err)
-	}
-
-	if errors.Is(err, backendplugin.ErrMethodNotImplemented) {
-		return response.Error(http.StatusNotFound, "Not found", err)
-	}
```

Comment:
Here 404 not found and in plugins/errors.go 501 not implemented. Need to decide which one it should be.

Reply:
good catch, I guess a 404 makes more sense since a plugin lacking a method (e.g. CallResource) is not a server error but a missing endpoint

Reply:
No biggy we can continue with not found. As I see it the resource (datasource or plugin) is found, but doesn't implement a certain method so maybe 501 would make more sense